docker/sui-node-deterministic/Dockerfile

```diff
@@ -60,11 +60,11 @@ ARG PROFILE
 ARG GIT_REVISION
 
 ENV RUST_BACKTRACE=1
-ENV RUSTFLAGS='-C target-feature=-crt-static -C codegen-units=1'
+ENV RUSTFLAGS='-C target-feature=+crt-static -C codegen-units=1'
 ENV GIT_REVISION=${GIT_REVISION}
 ENV PROFILE=${PROFILE}
 
-RUN --network=none cargo build --frozen --profile ${PROFILE} --bin sui-node
+RUN --network=none cargo build --target x86_64-unknown-linux-musl --frozen --profile ${PROFILE} --bin sui-node
```

Comment:
Is this part needed? or is the fixing of the rust flags enough?

Reply:
it actually is needed:
 ```
  => CACHED [build 7/8] COPY --from=linux-headers . /                                                                                                                                            0.0s
 => ERROR [build 8/8] RUN --network=none cargo build --frozen --profile release --bin sui-node                                                                                                  5.1s
------
 > [build 8/8] RUN --network=none cargo build --frozen --profile release --bin sui-node:
5.010 error: cannot produce proc-macro for `ark-ff-asm v0.4.2` as the target `x86_64-unknown-linux-musl` does not support these crate types
------
Dockerfile:67
--------------------
  65 |     ENV PROFILE=${PROFILE}
  66 |     
  67 | >>> RUN --network=none cargo build --frozen --profile ${PROFILE} --bin sui-node
  68 |     
  69 |     FROM scratch AS install
--------------------
ERROR: failed to solve: process "/bin/sh -c cargo build --frozen --profile ${PROFILE} --bin sui-node" did not complete successfully: exit code: 101
```
(other references to this issue of needing to specify `--target` 

https://github.com/rust-lang/rust/issues/78210#issuecomment-714776007

https://msfjarvis.dev/posts/building-static-rust-binaries-for-linux/
> The reason behind this seems to be a bug with cargo, where the RUSTFLAGS are applied to the target platform only when --target is explicitly provided